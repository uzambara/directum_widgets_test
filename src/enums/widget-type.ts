export enum WidgetType {
    Weather = 1,
    CurrencyRate = 2
}