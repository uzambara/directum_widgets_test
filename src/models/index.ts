import {INamedEntityWithId} from "./shared/named-entity-with-id";
import {ICity} from "./shared/city";
import {ICurrency} from "./currency-widget/currency";
import {IWidget} from "./shared/widget";
import {ICurrencyRateWidget} from "./currency-widget/currency-rate-widget";
import {IWeatherWidget} from "./weather-widget/weather-widget";
import {ICurrencyRateWidgetData} from "./currency-widget/currency-rate-widget-data";
import {IWeatherWidgetData} from "./weather-widget/weather-widget-data";

export {
    ICity,
    ICurrency,
    INamedEntityWithId,
    IWidget,
    ICurrencyRateWidget,
    IWeatherWidget,
    ICurrencyRateWidgetData,
    IWeatherWidgetData
}