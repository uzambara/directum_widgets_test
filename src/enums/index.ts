import {WidgetType} from "./widget-type";
import {WeatherType} from "./weather/weather-type";
import {PrecipitationType} from "./weather/precipitation-type"
import {CurrencyType} from "./currency/currency-type";
import {TemperatureUnitType} from "./weather/temperature-unit-type";


export {
    WidgetType,
    WeatherType,
    PrecipitationType,
    TemperatureUnitType,
    CurrencyType
};