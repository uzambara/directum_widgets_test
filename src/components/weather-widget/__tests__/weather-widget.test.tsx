import {mount, ReactWrapper} from "enzyme";
import {IWeatherWidgetProps, WeatherWidget} from "../../index";
import {IWeatherWidget, IWeatherWidgetData} from "../../../models";
import {TemperatureUnitType, WidgetType} from "../../../enums";
import React from "react";
import {WeatherServiceMock} from "../../../services/weather/__mocks__/weather.service.mock";
import {act} from "react-dom/test-utils";
import renderer from 'react-test-renderer';


const weatherServiceMock = new WeatherServiceMock();
const onEditWidget = jest.fn();
const onDeleteWidget = jest.fn();
const DEFAULT_WEATHER_WIDGET: IWeatherWidget = {
    city: {
        displayName: "displayName",
        id: "id",
        lat: "lat",
        lon: "lon"
    },
    temperatureUnit: TemperatureUnitType.Celsius,
    columnIndex: 0,
    id: "1",
    type: WidgetType.Weather
};

describe("weather widget tests", () => {
    let defaultWeatherWidget: IWeatherWidget;
    beforeEach(() => {
        jest.resetAllMocks();
        weatherServiceMock.getWeather = jest.fn().mockImplementation(() => {
            return new Promise(resolve => {
                let weatherWidgetData: IWeatherWidgetData = {
                    cityName: "cityName",
                    currentDate: "currentDate",
                    currentTime: "currentTime",
                    humidity: "humidity",
                    pressure: "pressure",
                    speedOfWind: "speedOfWind",
                    temperature: "temperature",
                    temperatureFeelsLike: "temperatureFeelsLike",
                    weatherIconUrl: "weatherIconUrl",
                    weatherType: "weatherType"

                };
                return resolve(weatherWidgetData);
            });
        });
        defaultWeatherWidget = DEFAULT_WEATHER_WIDGET;
    });

    it("render", () => {
        // Act
        const snapshot = renderer
            .create(<WeatherWidget
                widget={defaultWeatherWidget}
                weatherService={weatherServiceMock}
                onEditWidget={onEditWidget}
                onDeleteWidget={onDeleteWidget}
            />)
            .toJSON();

        // Asserts
        expect(snapshot).toMatchSnapshot();
    });

    it("первый рендер инициализирует данные виджета через сервис", async () => {
        // Arrange
        let component: ReactWrapper<IWeatherWidgetProps>;

        // Act
        await act(async () => {
            component = mount(<WeatherWidget
                widget={defaultWeatherWidget}
                weatherService={weatherServiceMock}
                onEditWidget={onEditWidget}
                onDeleteWidget={onDeleteWidget}
            />);
        });

        // Asserts
        expect(weatherServiceMock.getWeather).toHaveBeenCalledWith(defaultWeatherWidget);
    });
});