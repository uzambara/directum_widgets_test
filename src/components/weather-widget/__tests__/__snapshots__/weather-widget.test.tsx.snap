// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather widget tests render 1`] = `
<article
  className=""
  id="handle"
>
  <ul
    onClick={[Function]}
  >
    <li />
    <li />
    <li />
  </ul>
</article>
`;
